### ISSUE TITLE
LodePNG metadata compression compatibility problems

### ISSUE THREAD

**Issue by boardhead** [FLIF-hub/FLIF#307]
I get a "data error" when I try to decompress FLIF XMP metadata using the standard Zlib deflate.  Round trip converting the FLIF to PNG and back again is fine, so at least the LodePNG routines are self consistent.  However, the compressed data is different in the compressed PNG iTXt chunk than it is in the FLIF file.  It is not just a simple difference in compression level either -- I tried changing this and can't duplicate the compressed data in the FLIF file with Zlib deflate.

The bottom line is that the FLIF deflated metadata doesn't seem to be compatible with Zlib somehow.  This must be resolved for the metadata to be useful.

**Comment by jonsneyers** [comment]
I suppose this is the same issue as #308, right?

**Comment by boardhead** [comment]
Thanks for responding.

No, this is a different issue.  But I just found the reason:

https://github.com/FLIF-hub/FLIF/commit/e57098304ae20f71018cc07a9fede6b6b5054772

The data is purposely not Zlib compatible to save the 6 bytes and checksum calculation.  This is unfortunate, but I can deal with it.

So you can mark this issue as solved.

**Comment by jonsneyers** [comment]
Ah yes, those 6 bytes :)
If you just want to extract the metadata, you can do e.g. `flif input.flif output.xmp` and it will do what you expect it to do. The output file will be uncompressed.

**Comment by boardhead** [comment]
Thanks, but I was writing my own FLIF metadata decoder.

ExifTool 10.30 (just released) has the ability to read FLIF metadata:  http://owl.phy.queensu.ca/~phil/exiftool/

**Comment by jonsneyers** [comment]
Cool! Plans to add write support too?

**Comment by boardhead** [comment]
No plans for write support at the moment -- I'll wait until I get some requests first.

**Comment by boardhead** [comment]
Scratch that.

ExifTool 10.31 will have write support for FLIF metadata.

(The FLIF file format is so easy to work with, and I had a couple of free hours.)

**Comment by jonsneyers** [comment]
Heh, cool! Glad you consider the format to be easy to work with, that's good to know :)

**Comment by boardhead** [comment]
ExifTool 10.31 is now available.
